avocado/plugins/html.py

```diff
@@ -15,13 +15,277 @@
+                        os.path.basename(source))
+                    pkg_resources.ensure_directory(dest)
+                    shutil.copy(source, dest)
+
+    @staticmethod
+    def _open_browser(html_path):
+        # if possible, put browser in separate process
+        # group, so keyboard interrupts don't affect
+        # browser as well as Python
+        setsid = getattr(os, 'setsid', None)
+        if not setsid:
+            setsid = getattr(os, 'setpgrp', None)
+        inout = file(os.devnull, "r+")
+        cmd = ['xdg-open', html_path]
+        subprocess.Popen(cmd, close_fds=True, stdin=inout,
+                         stdout=inout, stderr=inout,
+                         preexec_fn=setsid)
+
+    def _render(self, result, output_path):
+        context = ReportModel(result=result, html_output=output_path)
+        template = pkg_resources.resource_string(
+            'avocado.core',
+            'resources/htmlresult/templates/report.mustache')
+
+        # pylint: disable=E0611
+        try:
+            if hasattr(pystache, 'Renderer'):
+                renderer = pystache.Renderer('utf-8', 'utf-8')
+                report_contents = renderer.render(template, context)
+            else:
+                from pystache import view
+                v = view.View(template, context)
+                report_contents = v.render('utf8')  # encodes into ascii
+                report_contents = codecs.decode("utf8")  # decode to unicode
+        except UnicodeDecodeError as details:
+            # FIXME: Remove me when UnicodeDecodeError problem is fixed
+            import logging
+            ui = logging.getLogger("avocado.app")
+            ui.critical("\n" + ("-" * 80))
+            ui.critical("HTML failed to render the template: %s\n\n",
+                        template)
+            ui.critical("-" * 80)
+            ui.critical("%s:\n\n", details)
+            ui.critical("%r", getattr(details, "object", "object not found"))
+            ui.critical("-" * 80)
+            raise
+
+        self._copy_static_resources(output_path)
+        with codecs.open(output_path, 'w', 'utf-8') as report_file:
+            report_file.write(report_contents)
+
+    def render(self, result, job):
+        if not (hasattr(job.args, 'html_job_result') or
+                hasattr(job.args, 'html_output')):
+            return
+
+        open_browser = getattr(job.args, 'open_browser', False)
+        if getattr(job.args, 'html_job_result', 'off') == 'on':
+            html_dir = os.path.join(job.logdir, 'html')
+            os.makedirs(html_dir)
+            html_path = os.path.join(html_dir, 'results.html')
+            self._render(result, html_path)
+            if getattr(job.args, 'stdout_claimed_by', None) is None:
+                log = logging.getLogger("avocado.app")
+                log.info("JOB HTML   : %s", html_path)
+            if open_browser:
+                self._open_browser(html_path)
+                open_browser = False
```

Comment:
I think you missed https://github.com/avocado-framework/avocado/pull/1372#discussion_r74436953 the commit message says it's not changing anything, but it does... In fact the proper solution would be to only to move the code and fix it in the following patch, but as the whole approach changed I'm not against bundling this fix here if the commit message describes the change.